### ISSUE TITLE
Direct control using CURL, Python or JavaScript ?

### ISSUE THREAD

**Issue by oywino** [themoosman/blynclight-api-server#4]
I'd like to be able to control my Blynclight from Home Assistant using either CURL, Python or JavaScript.
I do not with to use my headset or camera as event drivers. Any event that can occur on the Home Assistant Event bus should be usable as a triggers for various Blynclight controls through Automations.
Basically, all I need to know is what to send out on the USB port for the various Blynclight features, and how to accomplish sending it.

Can your API be used for this purpose?

**Comment by themoosman** [comment]
Funny you ask.  I recently did this, so now HA controls the light.  Let me get that together and update the repo.

**Comment by oywino** [comment]
Wow! Great 👍

**Comment by oywino** [comment]
Any progress ?  😊

**Comment by themoosman** [comment]
I'll be posting a brand new repo shortly.  This was a complete rewrite to use MQTT.

**Comment by oywino** [comment]
Superb! 😊🎇
New years gift 🎶

**Comment by themoosman** [comment]
@oywino Have a look:  https://github.com/themoosman/busylight-mqtt/tree/feature/init

**Comment by oywino** [comment]
Exiting! But before diving into it - have some uncertainties;
My MQTT Server is the official HA version: https://www.home-assistant.io/integrations/mqtt

1.  The `BusyLight for Humans` Python package offers two different installation methods. Which one should be used?
2.  In any case, where should the package be installed? (My HA instance runs in a Docker Container on a NAS)
I can see three possible routes:
a) Install inside the HA container (not recommended as it will not survive HA updates)
b) Install it in a separate container
c) install in directly on the NAS using a Python virtual environment
What do you recommend?

**Comment by themoosman** [comment]
1) Good point; the doc is misleading.  The requirements.txt file has the dependency in it.  I updated the README
2) it would have to be `c` until I get this containerized.  I'm working on it now, but having an issue passing the USB device into the container.

**Comment by oywino** [comment]
Agree. passing hardware level access (like USB) to a container requires running the container in privilege mode which leaves it suspectable to cause host system havoc.

**Comment by oywino** [comment]
Oh, I just realized - my Blynclight is connected to a Windows client PC, and not to the NAS where HA runs.  I guess that changes things a bit for my case?

**Comment by oywino** [comment]
I could use some advise on how to progress, please?

**Comment by themoosman** [comment]
Get a Linux machine.  The underlying library only works on Mac or Linux.

**Comment by oywino** [comment]
HA itself is running on a Linux machine and my intention was to install your "add-on" in a container, also on a Linux machine, but these servers are located in a closed, dark and cold server room. I have no need for a Blynclight in there. I need them attached to client PC's

**Comment by themoosman** [comment]
Buy an old cheap Raspberry Pi 2/3 to run the python and control the light.

**Comment by themoosman** [comment]
If you have any additional questions, move them to the new repo.
